src/main/docker/bamboo/c/Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM atlassian/bamboo-server:7.2.3
+
+LABEL description="Bamboo pre-configured for Artemis and C"
+
+USER root
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    software-properties-common=0.9* \
+    apt-utils=2.0.* \
+    maven=3.6.* \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+RUN add-apt-repository -y ppa:linuxuprising/java
+RUN echo debconf shared/accepted-oracle-license-v1-2 select true | debconf-set-selections && echo debconf shared/accepted-oracle-license-v1-2 seen true | debconf-set-selections
+RUN apt-get install -y --no-install-recommends oracle-java16-installer=16.*
+RUN update-alternatives --install /usr/bin/java java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java 1
+RUN update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
+# Add file in /artemis/bin/mvn that uses the correct java version and passes all arguments to mvn
+RUN mkdir /artemis && mkdir /artemis/bin && printf '#!/bin/bash\nJAVA_HOME=/usr/lib/jvm/java-16-oracle /usr/bin/mvn "$@"\n' > /artemis/bin/mvn && chmod 777 /artemis/bin/mvn
+
+# Install dependencies for C static analysis
+RUN apt-get install -y --no-install-recommends cppcheck=1.9* gcc-10=10.3* python3=3.8* python3-pip=20.0* sudo=1.8*
```

Comment:
Since it is only an additional compiler flag, normal compilation warnings are also present in the GCC report. 
Take a look at the example output [here](https://github.com/ls1intum/static-code-analysis-parser/blob/a1da707e9200fd3fefe2a4234cf27ca2dca74980/src/test/java/gcc.xml). The report still contains things like `[-Wimplicit-function-declaration]`, which are just normal GCC warnings not related to SCA. So `[-Wimplicit-function-declaration]` e.g. would be put in the `Miscellaneous` category.
However, it might be a good idea to remove the warning flags from the `Makefile` target. But there should still be warnings that the GCC outputs by default and also the possibility that the code won't even compile, which would result in errors that can be found under `Miscellaneous`.
Also, keep in mind that this Docker file is not used in production but rather for local testing.

Reply:
That sounds reasonable.